qiskit_dynamics/signals/__init__.py

```diff
@@ -12,14 +12,195 @@
+
+
+Constant Signal
+^^^^^^^^^^^^^^^
+
+:class:`~qiskit_dynamics.signals.Signal` supports specification of a *constant signal*:
+
+.. code-block:: python
+
+    const = Signal(2.)
+
+This initializes the object to always return the constant ``2.``, and allows constants to be
+treated on the same footing as arbitrary :class:`~qiskit_dynamics.signals.Signal` instances.
+A :class:`~qiskit_dynamics.signals.Signal` operating in constant-mode can be checked via the
+boolean attribute ``const.is_constant``.
+
+
+DiscreteSignal
+^^^^^^^^^^^^^^
+
+A :class:`~qiskit_dynamics.signals.DiscreteSignal`  is specified
+as an array of samples with a sample width and an optional start point (defaults to ``0.``).
+:class:`~qiskit_dynamics.signals.DiscreteSignal` inherits the same API as
+:class:`~qiskit_dynamics.signals.Signal`, and once instantiated can be interacted with
+in the same way.
+
+.. jupyter-execute::
+
+    from qiskit_dynamics.signals import DiscreteSignal
+    from matplotlib import pyplot as plt
+
+    discrete_signal = DiscreteSignal(dt=0.5, samples=[1., 2., 3.], carrier_freq=1.)
+
+    fig, axs = plt.subplots(1, 2, figsize=(14, 4))
+    discrete_signal.draw(t0=0., tf=1.5, n=100,
+                         axis=axs[0], title='Signal')
+    discrete_signal.draw(t0=0., tf=1.5, n=100,
+                         function='complex_value', axis=axs[1], title='Complex value')
+
+Note that the carrier wave is still analog for :class:`~qiskit_dynamics.signals.DiscreteSignal`.
+
+Algebraic Operations and composite signals
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Signals support algebraic operations. Adding two signals results in a
+:class:`~qiskit_dynamics.signals.SignalSum` object:
+
+.. code-block:: python
+
+    signal_sum = signal1 + signal2
+
+
+:class:`~qiskit_dynamics.signals.SignalSum` again implements the same API
+as :class:`~qiskit_dynamics.signals.Signal`, so for many purposes it can be
+interacted with in the same manner as :class:`~qiskit_dynamics.signals.Signal`
+and :class:`~qiskit_dynamics.signals.DiscreteSignal`.
+
+The components of a sum are stored in the ``components`` attribute as a list, but may
+also be accessed via subscripting, e.g. ``signal_sum[0]`` will produce ``signal1`` in
+the codeblock above.
+
+Signals of any type can be added together in this way. However, in the special case that
+:class:`~qiskit_dynamics.signals.DiscreteSignal`\s with compatible sample structure
+(same number of samples, ``dt``, and start time) are added together,
+a :class:`~qiskit_dynamics.signals.DiscreteSignalSum` is produced.
+:class:`~qiskit_dynamics.signals.DiscreteSignalSum` stores a sum of compatible
+:class:`~qiskit_dynamics.signals.DiscreteSignal`\s by joining the underlying arrays,
+so that the sum can be evaluated using purely array-based operations.
+
+Signal multiplication is also supported via :class:`~qiskit_dynamics.signals.SignalSum`
+and :class:`~qiskit_dynamics.signals.DiscreteSignalSum` using the identity:
```

Comment:
There should be a summary table somewhere (probably at the start) of this doc that lists all the signal classes with a short description of what they represent.

Reply:
See previous comment.